src/array/kernel.cc

```diff
@@ -235,6 +326,52 @@ DGL_REGISTER_GLOBAL("sparse._CAPI_DGLKernelSpMM")
     SpMM(op, reduce_op, graph.sptr(), U, E, V, {ArgU, ArgE});
   });
 
+DGL_REGISTER_GLOBAL("sparse._CAPI_DGLKernelSpMMHetero")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    HeteroGraphRef graph = args[0];
+    const std::string op = args[1];
+    const std::string reduce_op = args[2];
+    List<Value> list_U = args[3];
+    List<Value> list_E = args[4];
+    List<Value> list_V = args[5];
+    NDArray ArgU = args[6];
+    NDArray ArgE = args[7];
+    std::vector<NDArray> U_vec;
+    std::vector<NDArray> V_vec;
+    std::vector<NDArray> E_vec;
+    // CHECK_EQ(list_U.size(), list_V.size());
+    U_vec.reserve(list_U.size());
+    V_vec.reserve(list_V.size());
+    E_vec.reserve(list_E.size());
+    for (Value val : list_U) {
+      U_vec.push_back(val->data);
+    }
+    for (Value val : list_V) {
+      V_vec.push_back(val->data);
+    }
+    for (Value val : list_E) {
+      E_vec.push_back(val->data);
+    }
+
+    for (dgl_type_t etype = 0; etype < graph->NumEdgeTypes(); ++etype) {
```

Comment:
this format is to verify the metadata of the input graph?

Reply:
Do you mean why we are converting from list to vector?